picard/acoustid/__init__.py

```diff
@@ -106,12 +106,18 @@ def _on_lookup_finished(self, next_func, file, document, http, error):
                                 parsed_recording['score'] = score * result_score
                                 parsed_recording['acoustid'] = result['id']
                                 recording_list.append(parsed_recording)
-                        log.debug("AcoustID: Lookup successful for '%s'", file.filename)
 
-                    # Set AcoustID in tags if there was no matching recording
-                    if results and not recording_list:
-                        file.metadata['acoustid_id'] = results[0]['id']
-                        file.update()
+                    if results:
+                        if not recording_list:
+                            # Set AcoustID in tags if there was no matching recording
+                            file.metadata['acoustid_id'] = results[0]['id']
+                            file.update()
+                        else:
```

Comment:
I guess we should have some debug output here as well. Something like "AcoustID: Found for '%s', but recording matches below threshold".

Reply:
Done, and rebased